app/assets/javascripts/spotlight/admin/crop.es6

```diff
@@ -0,0 +1,315 @@
+    this.iiifUrlField.val(source);
+    this.setupIiifCropper();
+  }
+
+  // Render the Leaflet Map into the crop area
+  renderCropperMap() {
+    if (this.cropperMap) {
+      return;
+    }
+    this.cropperMap = L.map(this.cropArea.attr('id'), {
+      editable: true,
+      center: [0, 0],
+      crs: L.CRS.Simple,
+      zoom: 0,
+      editOptions: {
+        rectangleEditorClass: this.aspectRatioPreservingRectangleEditor(this.aspectRatio())
+      }
+    });
+    this.invalidateMapSizeOnTabToggle();
+  }
+
+  // Render the crop box (a Leaflet editable rectangle) onto the canvas
+  renderCropBox(initialBounds) {
+    this.cropBox = L.rectangle(initialBounds);
+    this.cropBox.addTo(this.cropperMap);
+    this.cropBox.enableEdit();
+    this.cropBox.on('dblclick', L.DomEvent.stop).on('dblclick', this.cropBox.toggleEdit);
+
+    var self = this;
+    this.cropperMap.on('editable:dragend editable:vertex:dragend', function(e) {
+      var bounds = e.layer.getBounds();
+      var region = self.projectBoundsToIIIFRegion(bounds);
+
+      self.iiifRegionField.val(region.join(','));
+    });
+  }
+
+  // Get the maximum zoom level for the IIIF Layer (always 1:1 image pixel to canvas?)
+  maxZoom() {
+    if(this.imageLayer) {
+      return this.imageLayer.maxZoom;
+    }
+  }
+
+  // Take a Leaflet LatLngBounds object and transform it into a IIIF [x, y, w, h] region
+  projectBoundsToIIIFRegion(bounds) {
+    var min = this.cropperMap.project(bounds.getNorthWest(), this.maxZoom());
+    var max = this.cropperMap.project(bounds.getSouthEast(), this.maxZoom());
+    return [
+      Math.max(Math.floor(min.x), 0),
+      Math.max(Math.floor(min.y), 0),
+      Math.floor(max.x - min.x),
+      Math.floor(max.y - min.y)
+    ];
+  }
+
+  // Take a IIIF [x, y, w, h] region and transform it into a Leaflet LatLngBounds
+  unprojectIIIFRegionToBounds(region) {
+    var minPoint = L.point(parseInt(region[0]), parseInt(region[1]));
+    var maxPoint = L.point(parseInt(region[0]) + parseInt(region[2]), parseInt(region[1]) + parseInt(region[3]));
+
+    var min = this.cropperMap.unproject(minPoint, this.maxZoom());
+    var max = this.cropperMap.unproject(maxPoint, this.maxZoom());
+    return L.latLngBounds(min, max);
+  }
+
+  // TODO: Add accessors to update hidden inputs with IIIF uri/ids?
+
+  // Setup autocomplete inputs to have the iiif_cropper context
+  setupAutoCompletes() {
+    var input = $('[data-behavior="autocomplete"]', this.cropTool);
+    input.data('iiifCropper', this);
+  }
+
+  setupAjaxFileUpload() {
+    this.fileInput = $('input[type="file"]', this.cropTool);
+    this.fileInput.change(() => this.uploadFile());
+  }
+
+  addImageSelectorToExistingCropTool() {
+    if(this.iiifManifestField.val() === '') {
+      return;
+    }
+  
+    var input = $('[data-behavior="autocomplete"]', this.cropTool);
+    var panel = $(input.data('target-panel'));
+    // This is defined in search_typeahead.js
+    // BEGIN CUSTOMIZATION
```

Comment:
Seems like there should be a way to monkeypatch just this addImageSelectorToExistingCropTool function (with something like `Crop.prototype.addImageSelectorToExistingCropTool`) instead of copying over this whole file, but I wasn't able to figure out how to properly import Crop from the spotlight gem to add to prototype. Super welcome to ideas.

Reply:
Possibly related to how spotlight loads es6 modules. Seems like this will be more straightforward once we upgrade to latest spotlight, which changes how js assets are compiled.